compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -338,5 +346,20 @@ class ReifyQuotes extends MacroTransformWithImplicits {
             checkLevel(mapOverTree(enteredSyms))
         }
       }
+
+    /** InlineSplice is used to detect cases where the expansion
+     *  consists of a (possibly multiple & nested) block or a sole expression.
+     */
+    object InlineSplice {
+      def unapply(tree: Tree)(implicit ctx: Context): Option[Select] = {
+        tree match {
+          case expansion: Select if expansion.symbol.isSplice =>
+            Some(expansion)
+          case Block(List(stat), Literal(Constant(()))) => unapply(stat)
```

Comment:
Can you give an example of when this happen?

Reply:
Yes, this is related to the implicit conversion related to value discarding (simply put when an `e` returns a value and it's called from a function of expected type `Unit` and it is converted into `{ e; () }`). This manifested in quoted code as well.

Reply:
For example `inline foo: Unit = 4` will become `inline foo: Unit = { 4; () }` and then be inlined as `{ 4; () }`. In our case we have a `~xyz` instead of `4`.

Reply:
Got it! I would add a comment explaining why this is needed (i.e. the compiler can insert `()`).